crates/ethereum/evm/src/eip6110.rs

```diff
@@ -0,0 +1,75 @@
+//! EIP-6110 deposit requests parsing
+use alloy_consensus::Request;
+use alloy_eips::eip6110::{DepositRequest, MAINNET_DEPOSIT_CONTRACT_ADDRESS};
+use alloy_sol_types::{sol, SolEvent};
+use reth_interfaces::executor::BlockValidationError;
+use reth_primitives::Receipt;
+use revm_primitives::Log;
+
+pub(crate) fn parse_deposits_from_receipts(
+    receipts: &[Receipt],
+) -> Result<Vec<Request>, BlockValidationError> {
+    let res = receipts
+        .iter()
+        .flat_map(|receipt| receipt.logs.iter())
+        // No need to filter for topic because there's only one event and that's the Deposit event
+        // in the deposit contract.
+        .filter(|log| log.address == MAINNET_DEPOSIT_CONTRACT_ADDRESS)
+        .map(|log| DepositEvent::decode_log(log, false))
+        .map(|res| {
+            let deposit = parse_deposit_from_log(&res?);
+            Ok(Request::DepositRequest(deposit))
+        })
+        .collect::<Result<Vec<_>, _>>()
+        // todo: this is ugly, we should clean it up
+        .map_err(|err: alloy_sol_types::Error| {
+            BlockValidationError::DepositRequestDecode(err.to_string())
+        })?;
+    Ok(res)
+}
+
+sol! {
+    #[allow(missing_docs)]
+    event DepositEvent(
+        bytes pubkey,
+        bytes withdrawal_credentials,
+        bytes amount,
+        bytes signature,
+        bytes index
+    );
+}
+
+fn parse_deposit_from_log(log: &Log<DepositEvent>) -> DepositRequest {
```

Comment:
maybe we should put the `DepositEvent` in alloy-eips for the relevant eip, and have a conversion from `DepositEvent → DepositRequest` in alloy

Reply:
agreed, let's do it after